src/Hedgehog.Experimental.CSharp.Tests/GenTests.cs

```diff
@@ -0,0 +1,21 @@
+namespace Hedgehog.Experimental.CSharp.Tests;
+
+using Hedgehog.Linq;
+using FluentAssertions;
+
+public record MyRecord(string Name, int Age);
+
+public class GenTests
+{
+  [Fact]
+  public void ShouldUseGenConfig()
+  {
+    var onlyString = "singleton string";
+    var config = GenX.Defaults.WithGenerator(Gen.FromValue(onlyString));
```

Comment:
I guess this is because both `Hedgehog` and `Hedgehog.Linq` define type called `Gen`.
This is super confusing because `Gen` in `Hedgehog` is the actual generator, while `Gen` in `Hedgehog.Linq` is a "static class" with helper C#-friendly methods in it.

C# also seems to resolves types relatively to the current namespace hierarchy. 
So when you do `namespace` after `using`, your `namespace` seems to be "closer" scope and it searches the type `Gen` in the namespace, traversing up in its hierarchy. And it finds `Gen` in `Hedgehog` before it explores "global" usings.

Having different things named the same (`Gen`) in two different namespaces when you practically need both to do something more or less realistic is super confusing.

Reply:
Sounds about right to me: https://stackoverflow.com/a/151560

Reply:
Yes, I understand this now.  Thanks to both of you.

So we give them different names, right?

How about we rename `Hedgehog.Linq.Gen` to `Hedgehog.Linq.GenL`?

Reply:
In `Hedgehog`? 
I _think_ that the effort there is to make `Hedgehog.Linq` to be self-sufficient. So C# devs would just reference that namespace and never resort to `Gen` in `Hedgehog`. 

That's fair.
But the fact that `Gen` exists in both namespaces, IDE sometimes gets confused and the mess starts.

TBH, because C# devs should probably never need to access `Hedgehog.Gen` I would use the same trick that F# itself uses for `Option`, `Result`, etc. and added `CompileName` attribute on `Hedgehog.Gen`.

This way in F# everything would be as it is, but from C# `Hedgehog.Gen` type would have a different name and there won't be any confusion.

But I think that having `GenL` in API isn't the nicest user experience...

Reply:
> I _think_ that the effort there is to make `Hedgehog.Linq` to be self-sufficient. So C# devs would just reference that namespace and never resort to `Gen` in `Hedgehog`.

Oh, yes.  That is correct.  I had forgot that when I was working with you code earlier today.

> TBH, because C# devs should probably never need to access `Hedgehog.Gen` I would use the same trick that F# itself uses for `Option`, `Result`, etc. and added `CompileName` attribute on `Hedgehog.Gen`.
> 
> This way in F# everything would be as it is, but from C# `Hedgehog.Gen` type would have a different name and there won't be any confusion.

IIRC, we introduced this split after support for Fable was added.  Fable can't handle the `CompileName` attribute, so we had to wrap every `CompileName` attribute in compiler directives.  It made the code very verbose.  I found it very hard to "just" read the underlying F# code.

> But I think that having `GenL` in API isn't the nicest user experience...

I don't like this either.  Just threw out the idea to see what you thought.